libs/template/renderer.go

```diff
@@ -112,6 +119,17 @@ func (r *renderer) executeTemplate(templateDefinition string) (string, error) {
 	result := strings.Builder{}
 	err = tmpl.Execute(&result, r.config)
 	if err != nil {
+		// Parse and return a more readable error for missing values that are used
+		// by the template definition but are not provided in the passed config.
+		target := &template.ExecError{}
+		if errors.As(err, target) {
+			captureRegex := regexp.MustCompile(`map has no entry for key "(.*)"`)
+			matches := captureRegex.FindStringSubmatch(target.Err.Error())
+			if len(matches) != 2 {
+				return "", err
+			}
+			return "", fmt.Errorf("variable %q not defined", matches[1])
```

Comment:
Please return a new `template.ExecError` where you forward the `Name`.

Reply:
Also, I think you can use https://pkg.go.dev/regexp#Regexp.ReplaceAllString here (not blocking).

Reply:
Changed to return a ExecError. Note that today the Name is `""`. It's set by us. One optimization we can do is name it the file name, and then use the `ReplaceAllString` method. The advantage this has, is it also will have the file name and line number where the error happened.

Not high priority IMO, so we can skip it. FYI, The error then would look like:
```
"template: foo.tmpl:1:2: executing \"foo.tmpl\" at <.does_not_exist>: varible \"does_not_exist\" not defined"
```

The existing error though is good enough.

Reply:
How is the `Name` field related to `ReplaceAllString`?

Reply:
I meant to use `ReplaceAllString` on the string you're working with now, not the full error.

Then you don't have to check the length of matches, synthesize a new one, etc.

Reply:
> How is the Name field related to ReplaceAllString?
I meant we could specify a use the file name as the name for the template. Then during execution we could use ReplaceAllString to just replace "map has no entry for key ..." keeping the prefix of the error same. But yeah, existing approach is good enough.